corehq/apps/es/client.py

```diff
@@ -151,9 +151,15 @@ def get_task(self, task_id):
         :returns: ``dict`` of task details
         :raises: ``TaskError`` or ``TaskMissing`` (subclass of ``TaskError``)
         """
-        # NOTE: elasticsearch5 python library doesn't support `task_id` as a
-        # kwarg for the `tasks.list()` method, and uses `tasks.get()` for that
-        # instead.
+        if self.elastic_major_version == 5:
+            try:
+                task_details = self._es.tasks.get(task_id=task_id)
+                task_info = task_details['task']
+                task_info['completed'] = task_details['completed']
+            except NotFoundError as e:
+                # unknown task id provided
+                raise TaskMissing(e)
+            return task_info
         return self._parse_task_result(self._es.tasks.list(task_id=task_id,
                                                            detailed=True))
```

Comment:
Thanks @esoergel for having a look at the PR and apologies for the delayed response.

> What about the rest of the stuff that this did? Was tasks.get not available until ES5 or something?

Yeah. `tasks.get` was not available until ES 5 and `tasks.list` used to wrap a lot of extra information along with the task detaisl.

`_parse_task_result` majorly was responsible to extract out that information from the result body. And the error handling is used to determine if a task is finished. Until ES 5, the tasks api did not tell if a task is completed or errored. As soon as the task is finished(in any state), it used to be removed from tasks. So we had no successful way of determining if task is completed. We relied on the error to state that the task is missing which is now solved in ES 5. Now ES 5 returns a completed flag and task is kept in results for sometime after it is completed.

So it changes the entire way in which we used to consume the API hence the difference in output formats. It would be more streamlined going to newer versions. And I like your idea of having a named tuple for result which I will implement in a follow-up PR soon.